ndcube/meta.py

```diff
@@ -0,0 +1,374 @@
+        del self[name]
+
+    def __setitem__(self, key, val):
+        axis = self.axes.get(key, None)
+        if axis is not None:
+            if _not_scalar(val):
+                axis_shape = tuple(self.shape[axis])
+                if not _is_grid_aligned(val, axis_shape) and not _is_axis_aligned(val, axis_shape):
+                    raise TypeError(
+                        f"{key} is already associated with axis/axes {axis}. val must therefore "
+                        f"must either have same length as number associated axes ({len(axis)}), "
+                        f"or the same shape as associated data axes {tuple(self.shape[axis])}. "
+                        f"val shape = {val.shape if hasattr(val, 'shape') else (len(val),)}\n"
+                        "We recommend using the 'add' method to set values.")
+        super().__setitem__(key, val)
+
+    def __getitem__(self, item):
+        # There are two ways to slice:
+        # by key, or
+        # by typical python numeric slicing API,
+        # i.e. slice the each piece of metadata associated with an axes.
+
+        if isinstance(item, str):
+            return super().__getitem__(item)
+
+        elif self.shape is None:
+            raise TypeError("Meta object does not have a shape and so cannot be sliced.")
+
+        else:
+            new_meta = copy.deepcopy(self)
+            if isinstance(item, (numbers.Integral, slice)):
+                item = [item]
+            naxes = len(self.shape)
+            item = np.array(list(item) + [slice(None)] * (naxes - len(item)),
+                            dtype=object)
+
+            # Edit data shape and calculate which axis will be dropped.
+            dropped_axes = np.zeros(naxes, dtype=bool)
+            new_shape = new_meta.shape
+            for i, axis_item in enumerate(item):
+                if isinstance(axis_item, numbers.Integral):
+                    dropped_axes[i] = True
+                elif isinstance(axis_item, slice):
+                    start = axis_item.start
+                    if start is None:
+                        start = 0
+                    if start < 0:
+                        start = self.shape[i] - start
+                    stop = axis_item.stop
+                    if stop is None:
+                        stop = self.shape[i]
+                    if stop < 0:
+                        stop = self.shape[i] - stop
+                    new_shape[i] = stop - start
+                else:
+                    raise TypeError("Unrecognized slice type. "
+                                    "Must be an int, slice and tuple of the same.")
+            kept_axes = np.invert(dropped_axes)
+            new_meta._data_shape = new_shape[kept_axes]
+
+            # Slice all metadata associated with axes.
+            for key, value in self.items():
+                axis = self.axes.get(key, None)
+                drop_key = False
+                if axis is not None:
```

Comment:
I think it would be neater if this was:
```suggestion
                if axis is None:
                    continue
```